File: 405. Convert a Number to Hexadecimal/Python/Solution.py
class Solution(object):
    def toHex(self, num):
        """
        :type num: int
        :rtype: str
        """
        if num == 0:
            return '0'
        symbols = '0123456789abcdef'
        result = ''
        while num:
            value = num & 0xf
            result += symbols[value]
            num = num >> 4 if num > 0 else (num % 0x100000000) >> 4
        return result[::-1]

File: 405. Convert a Number to Hexadecimal/C++/Solution.cpp
class Solution {
public:
    string toHex(int num) {
        if (num == 0) {
            return "0";
        }
        string symbols = "0123456789abcdef";
        string result = "";
        while (num) {
            int value = num & 0xf;
            result += symbols[value];
            num = (int)((unsigned int) num >> 4);
        }
        reverse(result.rbegin(), result.rend());
        return result;
    }
};